libs/pilight/protocols/433.92/selectplus_doorbell.c

```diff
@@ -0,0 +1,183 @@
+/*
+	Copyright (C) 2020 woutput
+
+	This file is part of pilight.
+	This Source Code Form is subject to the terms of the Mozilla Public
+	License, v. 2.0. If a copy of the MPL was not distributed with this
+	file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+
+#include "../../core/pilight.h"
+#include "../../core/common.h"
+#include "../../core/dso.h"
+#include "../../core/log.h"
+#include "../protocol.h"
+#include "../../core/binary.h"
+#include "../../core/gc.h"
+#include "selectplus_doorbell.h"
+
+// Timing from TODO
+
+// Increase this value to be more robust, but also create more false positives
+#define PEAK_TO_PEAK_JITTER	80
+
+// The same jitter is assumed on every pulse type (as may be expected)
+
+// Short pulse timing
+#define MIN_SHORT_PULSE_LENGTH	(AVG_SHORT_PULSE_LENGTH - 0.5 * PEAK_TO_PEAK_JITTER)
+#define AVG_SHORT_PULSE_LENGTH	372
+#define MAX_SHORT_PULSE_LENGTH	(AVG_SHORT_PULSE_LENGTH + 0.5 * PEAK_TO_PEAK_JITTER)
+
+// Medium pulse timing
+#define MIN_MEDIUM_PULSE_LENGTH	(AVG_MEDIUM_PULSE_LENGTH - 0.5 * PEAK_TO_PEAK_JITTER)
+#define AVG_MEDIUM_PULSE_LENGTH	1094
+#define MAX_MEDIUM_PULSE_LENGTH	(AVG_MEDIUM_PULSE_LENGTH + 0.5 * PEAK_TO_PEAK_JITTER)
+
+// Long pulse timing
+#define MIN_LONG_PULSE_LENGTH	(AVG_LONG_PULSE_LENGTH - 0.5 * PEAK_TO_PEAK_JITTER)
+#define AVG_LONG_PULSE_LENGTH	6536
+#define MAX_LONG_PULSE_LENGTH	(AVG_LONG_PULSE_LENGTH + 0.5 * PEAK_TO_PEAK_JITTER)
+
+#define RAW_LENGTH		36
+// One pulse per bit, first 27 pulses are ID, last 9 pulses are footer
+#define BINARY_LENGTH		27
+
+#define NORMAL_REPEATS		68
```

Comment:
This much?

Reply:
You mean this many repeats? Yes, it's a lot, the doorbell button is sending for 2.2 seconds! See [protocol analysis attached to this post](https://forum.pilight.org/showthread.php?tid=1435&pid=25180#pid25180) on the forum.

Reply:
Didn't it work with less?

Reply:
I didn't try yet. 68 repeats is what the doorbell button does.
What is the intention of pilight in general? A) Mimic real devices or B) Minimize 433 MHz pollution, that is minimize transmit time, while being reliable or C) other?

Reply:
Be as good or better than the real device  So preferably B.

Reply:
OK, sounds good to me :)
I will do some testing.